dashboard/dashboard_functions.py

```diff
@@ -0,0 +1,150 @@
+"""This file contains the functions necessary to generate the Streamlit dashboard."""
+
+from os import environ
+from datetime import datetime, timedelta
+from dotenv import load_dotenv
+import pandas as pd
+import altair as alt
+from sqlalchemy import create_engine, URL
+from sqlalchemy.engine.base import Engine
+
+load_dotenv()
+
+engine_url_object = URL.create("postgresql+psycopg2",
+                               username=environ["DB_USER"],
+                               password=environ["DB_PASSWORD"],
+                               host=environ["DB_HOST"],
+                               database=environ["DB_NAME"])
+
+
+def make_clickable(url, name):
+    return '<a href="{}" rel="noopener noreferrer" target="_blank">{}</a>'.format(url,name)
+
+
+def generate_dataframe(engine: Engine, timeframe: str) -> pd.DataFrame:
+    """Returns a comprehensive dataframe containing information on
+    stories, records and topics."""
+    current_time = datetime.now()
+    if timeframe not in ("hour","day", "week"):
+        return None
+    if timeframe == "hour":
+        time_cutoff = current_time - timedelta(hours=1)
+    if timeframe == "day":
+        time_cutoff = current_time - timedelta(days=1)
+    if timeframe == "week":
+        time_cutoff = current_time - timedelta(days=7)
+    
+    query = f"""
+    SELECT records.record_id, records.story_id, records.score, records.comments, stories.title, stories.story_url, topics.name
+    FROM records
+    LEFT JOIN stories
+    ON records.story_id = stories.story_id
+    LEFT JOIN topics
+    ON stories.topic_id = topics.topic_id
+    WHERE record_time > '{time_cutoff}';"""
```

Comment:
You could put your queries into the sql_queries file I have used to make code more neater and concise.